alf/nest/nest.py

```diff
@@ -883,3 +883,36 @@ def set_field(nested, field, new_value):
     """
 
     return transform_nest(nested, field, lambda _: new_value)
+
+
+def transpose(nested, shallow_nest):
```

Comment:
I think `inside_out` might better describe the operation, if `transpose` is not a convention.

Reply:
> I think `inside_out` might better describe the operation, if `transpose` is not a convention.

In general, it can interchange any two "top levels", which is analogous to matrix transpose of two axes.